BUILD.md

```diff
@@ -0,0 +1,228 @@
+```
+
+Restart docker
+```bash
+$ sudo systemctl daemon-reload
+$ sudo systemctl restart docker
+```
+
+Verify that the configuration has been loaded:
+```bash
+$ systemctl show --property=Environment docker
+Environment=HTTP_PROXY=http://proxy-server.example:80/
+```
+
+**Docker DNS (Optional)**
+
+Docker build uses /etc/resolv.conf for setting up DNS servers for docker image
+builds. If you are receiving `Host not found` errors during docker build steps
+then you will need to add nameserver entries to the resolve.conf file. Edit the
+/etc/resolv.conf as root and add the dns servers for you network.
+
+For example to use Google's public DNS servers:
+```
+    nameserver 8.8.8.8
+    nameserver 8.8.4.4
+```
+
+
+Step Two: Clone The Repository
+-------------
+
+You'll need to have git installed in order to clone the Sawtooth source
+code repository. You can find up-to-date installation instructions
+at Getting Started - Installing Git: <https://git-scm.com/book/en/v2/Getting-Started-Installing-Git>.
+
+Open up a terminal and run the following:
+
+```bash
+   $ cd $HOME
+   $ mkdir sawtooth
+   $ cd sawtooth
+   $ git clone https://github.com/hyperledger/sawtooth-core.git
+```
+
+Step Three: Initializing Your Environment
+-------------
+
+The Sawtooth build and test infrastructure requires dependencies be built.
+These dependencies include the protocol buffers definitions (under the protos
+dir) be built for the target languages and a set of docker images be built with
+with the required build and runtime dependencies installed.  To build all the
+dependencies for all of the components of Sawtooth, run this command:
+
+```bash
+  $ bin/build_all
+```
+
+By default, this builds all the components for all the components and sdks in
+the repository. This is a very slow process. It is recommended that you only
+build the language dependencies you need for your development purposes. The
+minimum requirements for running the Sawtooth validator is the python build:
+
+```bash
+  $ bin/build_all -l python
+```
+
+To get details on all the options for the build_all script
+
+```bash
+  $ bin/build_all -h
+```
+
+If you are working on the core validator, only the ‘python’ language is
+required. If you are working on a particular language sdk then you will need to
+build the language that sdk is for as well.
+
+**Note:** This build environment uses docker to virtualize the build and
+execute the code in the development directory. This allows for modification
+made to the local source to be built and tested without installing local
+dependencies on your machine.
+
+**Note:**  If you wish to configure your development machine to do compilation
+directly on the host without a docker virtualization, the docker files in the
+sawtooth-core/docker. For example, the file sawtooth-core/docker
+sawtooth-dev-python describes the configuration and components needed to build
+and run the python components on a system.
```

Comment:
missing a couple words. I think it's supposed to be "If you wish to configure your development machine to do compilation directly on the host without ~a~ docker virtualization, ***see*** the docker files in the sawtooth-core/docker ***directory***. For example, the file **sawtooth-core/docker/sawtooth-dev-python** describes the configuration and components needed to build and run the python components on a system.